geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/strategies/SpatialFilterStrategy.scala

```diff
@@ -19,23 +19,24 @@ trait SpatialFilterStrategy[T, U] extends GeoMesaFeatureIndex[T, U] {
 
   import SpatialFilterStrategy.{StaticCost, spatialCheck}
 
-  def geom: String
+  // attributes are assumed to be a single geometry field
+  lazy private val Seq(geom) = attributes
```

Comment:
Is there any way that this is not true?

Reply:
I'm just wondering if there's any chance for an exception, etc.  (It seems believable that everything is ok here; just asking.)

Reply:
only if there's a new class that extends this trait and doesn't follow the current conventions... i thought about adding some validation here but seemed like overkill.